### PR TITLE
Ensure Tool Results are Added for Multiple Parallel Tool Calls

### PR DESCRIPTION
**Pull request by Rajaniraiyn**
This update ensures that all tool calls in an assistant message receive a corresponding result before performing a round trip. Without this fix, most API providers will throw an error due to missing results from parallel tool calls.

### CODE REVIEW

File: packages/react/src/runtimes/local/LocalThreadRuntime.tsx [hunk 1]
```diff
@@ -235,7 +233,12 @@ export class LocalThreadRuntime implements ThreadRuntime {
     };
     this.repository.addOrUpdateMessage(parentId, message);
 
-    if (added && shouldContinue(message)) {
+    // Check if all tool calls have results
+    const allToolCallsHaveResults = message.content.every(
+      (c) => c.type !== "tool-call" || c.result,
```
Comment (Yonom) [inline packages/react/src/runtimes/local/LocalThreadRuntime.tsx:238]: @Rajaniraiyn shouldContinue already checks for all tool calls having a result. this seems to be a duplicate check of the same thing

https://github.com/Yonom/assistant-ui/blob/c92547de23b878ec088f7c888dab4d1b3db1756e/packages/react/src/runtimes/local/shouldContinue.tsx#L6

File: packages/react/src/runtimes/local/LocalThreadRuntime.tsx [hunk 2]
```diff
@@ -213,13 +213,11 @@ export class LocalThreadRuntime implements ThreadRuntime {
     if (message.role !== "assistant")
       throw new Error("Tried to add tool result to non-assistant message");
 
-    let added = false;
```
Comment (Yonom) [inline packages/react/src/runtimes/local/LocalThreadRuntime.tsx:216]: Why is this added check removed?

It checks for repeat calls to addToolResult() to ensure that a roundtrip is only triggered if a tool result is added (and not updated)

**Review by Yonom — CHANGES_REQUESTED** [review]
I am not able to reproduce the bug you specify... 

Furthermore, this code removes the added check (why?) and the added check is already in shouldContinue (does that one not work?)

**Comment by Yonom** [comment]
There is a adjacent bug that I found while testing for this. 

If a message has pending tool calls; one can send new messages via the composer. I am fixing this in #644

**Comment by Rajaniraiyn** [comment]
Closing this PR as it has duplicate checks and there is a better way to do the tool execution via `useAssistantTool` instead of `useAssistantToolUI`

**Comment by Yonom** [comment]
@Rajaniraiyn thank you for the submission! I really appreciate your efforts :)

**Pull request closed without merging**